core/geometry/src/test/curve/CurveCurveCloseApproachXY.test.ts

```diff
@@ -885,6 +888,115 @@ describe("CurveCurveCloseApproachXY", () => {
+    ck: Checker,
+    allGeometry: GeometryQuery[],
+    geometryA: Arc3d,
+    geometryB: Arc3d,
+    numExpectedIntersections: number,
+    numExpectedPerpCloseApproach: number,
+  ) {
+    const existsInIntersectionSetWithTolerance = (set: Set<Point3d>, point: Point3d, tolerance: number = 1e-6): boolean => {
+      for (const pt of set)
+        if (pt.isAlmostEqual(point, tolerance))
+          return true;
+      return false;
+    };
+
+    const existsInApproachSetWithTolerance = (
+      set: Set<[Point3d, Point3d]>, point0: Point3d, point1: Point3d, tolerance: number = 1e-6,
+    ): boolean => {
+      for (const startEnd of set) {
+        const start = startEnd[0];
+        const end = startEnd[1];
+        if ((start.isAlmostEqual(point0, tolerance) && end.isAlmostEqual(point1, tolerance)) ||
+          (start.isAlmostEqual(point1, tolerance) && end.isAlmostEqual(point0, tolerance)))
+          return true;
+      }
+      return false;
+    };
+    const maxDistance = 50;
+    let dy = 0;
+    for (let angle = 0; angle < 360; angle += 10) {
+      GeometryCoreTestIO.captureCloneGeometry(allGeometry, geometryA, 0, dy);
+      const rotationAxis: Vector3d = Vector3d.create(0, 0, 1);
+      const rotationMatrix = Matrix3d.createRotationAroundVector(rotationAxis, Angle.createDegrees(angle))!;
+      const rotationTransform = Transform.createFixedPointAndMatrix(Point3d.create(0, 0, 0), rotationMatrix);
+      geometryB.tryTransformInPlace(rotationTransform);
+      GeometryCoreTestIO.captureCloneGeometry(allGeometry, geometryB, 0, dy);
+      const approaches = CurveCurve.closeApproachProjectedXYPairs(geometryA, geometryB, maxDistance);
+      let numUniqueIntersections = 0;
+      let numUniquePerpCloseApproach = 0;
+      const intersectionSet = new Set<Point3d>();
+      const approachSet = new Set<[Point3d, Point3d]>();
+      const approachLen = approaches.length;
+      ck.testLE(0, approachLen);
+      if (approachLen > 0) {
+        for (const ap of approaches) {
+          const start = ap.detailA.point;
+          const end = ap.detailB.point;
+          if (start.isAlmostEqual(end)) { // intersection between geometries
+            GeometryCoreTestIO.createAndCaptureXYCircle(allGeometry, start, 0.0625, 0, dy);
+            if (!existsInIntersectionSetWithTolerance(intersectionSet, start, 1e-6)) {
+              intersectionSet.add(start); // add unique intersections to set
+              numUniqueIntersections++;
+            }
+          } else { // closest approach between geometries
+            const approachSegment = LineSegment3d.create(start, end);
+            const lenSqr = start.distanceSquaredXY(end);
+            ck.testLE(
+              Math.sqrt(lenSqr), maxDistance, undefined, undefined, "approach length must be smaller than maxDistance",
+            );
+            const vec1 = Vector3d.createStartEnd(start, end);
+            const vec2 = geometryA.fractionToPointAnd2Derivatives(ap.detailA.fraction).vectorU;
```

Comment:
For future reference: you could get by with `fractionToPointAndDerivative` or `fractionToPointAndUnitTangent` here.

Reply:
I took `fractionToPointAnd2Derivatives` from one of my sandboxes but `fractionToPointAndDerivative` is more readable so I will use it.